llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -10614,13 +10614,10 @@ SDValue AArch64TargetLowering::LowerCTPOP_PARITY(SDValue Op,
     return SDValue();
 
   EVT VT = Op.getValueType();
-  if (VT.isScalableVector() ||
-      useSVEForFixedLengthVectorVT(VT, !Subtarget->isNeonAvailable()))
+  bool OverrideNEON = !Subtarget->isNeonAvailable() || VT.isFixedLengthVector();
+  if (VT.isScalableVector() || useSVEForFixedLengthVectorVT(VT, OverrideNEON))
```

Comment:
```suggestion
  if (VT.isScalableVector() || useSVEForFixedLengthVectorVT(VT, /*OverrideNEON=*/true))
```
Although this depends on my i16 based vector comment.